File: ccif/src/java/other/org/himalay/ccif/CpDataReqMsg.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:59 EST 2016")

public  class CpDataReqMsg extends BinStruct implements PublicBinMsg{




    // members variables
    // dataTypeId
    public short dataTypeId ;
    // dataType
    public ByteArray dataType ;
    // DataType_id
    public ByteArray DataType_id ;

        public CpDataReqMsg () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize dataTypeId
        
        // Initialize dataType
        dataType= new ByteArray();dataType.setSizeType("FIRST_UI8");
        // Initialize DataType_id
        DataType_id= new ByteArray();DataType_id.setSizeType("FIRST_UI8");
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read dataTypeId
        {dataTypeId=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read dataType
        {retVal+=dataType.read(istream); }
        // read DataType_id
        {retVal+=DataType_id.read(istream); }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        
        { /** fix dependent sizes for dataType**/  }
        { /** fix dependent sizes for DataType_id**/  }
    
                // write dataTypeId
        ostream.writeByte(dataTypeId); retVal +=1;
        // write dataType
        {retVal += dataType.write(ostream);}
        // write DataType_id
        {retVal += DataType_id.write(ostream);}
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("CpDataReqMsg\n");
    dc.increaseIndent();
        int retVal= 0;
        // write dataTypeId
        dc.indent();dc.getPs().println("dataTypeId="+dataTypeId+"(0x"+ Integer.toHexString(dataTypeId) + ")" );
        // write dataType
        dc.indent();dc.getPs().print("dataType: "+dataType.getSize()+"(0x"+Integer.toHexString(dataType.getSize())+")\n");this.dataType.dump(dc);
        // write DataType_id
        dc.indent();dc.getPs().print("DataType_id: "+DataType_id.getSize()+"(0x"+Integer.toHexString(DataType_id.getSize())+")\n");this.DataType_id.dump(dc);
dc.decreaseIndent();
        return retVal;
    }


        // Getter for dataTypeId
    //public short getDataTypeId()
    //{
    //    return dataTypeId ;
    //}

    
    // Setter for dataTypeId
    //public void setDataTypeId(short val)
    //{
    //    this.dataTypeId= val;
    //}
    // Getter for dataType
    //public ByteArray getDataType()
    //{
    //    return dataType ;
    //}

    
    // Setter for dataType
    //public void setDataType(ByteArray val)
    //{
    //    this.dataType= val;
    //}
    // Getter for DataType_id
    //public ByteArray getDataType_id()
    //{
    //    return DataType_id ;
    //}

    
    // Setter for DataType_id
    //public void setDataType_id(ByteArray val)
    //{
    //    this.DataType_id= val;
    //}


    public void setDataType(byte[] val)
    {
        this.dataType.setData(val);
    }
    
    
    public void setDataType_id(byte[] val)
    {
        this.DataType_id.setData(val);
    }
    
    
    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libccif/src/bp/CpDataReqMsg.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:15 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "CpDataReqMsg.h"

#ifdef org_himalay_ccif_CpDataReqMsg__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
CpDataReqMsg::CpDataReqMsg() {
    references= 0;


    // dataTypeId
    dataTypeId= (ui8)0;
    // dataType
    org_himalay_msgs_runtime_ByteArray_NEW(dataType);
    dataType->setSizeType("FIRST_UI8");
    // DataType_id
    org_himalay_msgs_runtime_ByteArray_NEW(DataType_id);
    DataType_id->setSizeType("FIRST_UI8");
}
// Destructor
CpDataReqMsg::~CpDataReqMsg() {
    // dataTypeId

    // dataType
    org_himalay_msgs_runtime_ByteArray_DELETE( dataType);
    // DataType_id
    org_himalay_msgs_runtime_ByteArray_DELETE( DataType_id);
}

int CpDataReqMsg::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // dataTypeId
    {
        dataTypeId=(istream.readUI8());
        retVal+=1;
    }
    // dataType
    {
        retVal+=dataType->read(istream);
    }
    // DataType_id
    {
        retVal+=DataType_id->read(istream);
    }
    return retVal;
}

int CpDataReqMsg::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read dataTypeId
    {
        dataTypeId=(istream.readUI8());
        retVal+=1;
    }
    // read dataType
    {
        retVal+=dataType->read(istream);
    }
    // read DataType_id
    {
        retVal+=DataType_id->read(istream);
    }

    return retVal;
}

int CpDataReqMsg::write(DataOutputStream& ostream) { // throws IOException

    {   /** fix dependent sizes for dataType**/
    }
    {   /** fix dependent sizes for DataType_id**/
    }

    int retVal= 0;
    // write dataTypeId
    ostream.writeUI8(dataTypeId);
    retVal +=1;
    // write dataType
    {
        retVal += dataType->write(ostream);
    }
    // write DataType_id
    {
        retVal += DataType_id->write(ostream);
    }

    return retVal;
}

int CpDataReqMsg::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("CpDataReqMsg\n");
    dc.increaseIndent();
    int retVal= 0;
    // write dataTypeId
    dc.indent();
    dc.getPs().print("dataTypeId=");
    dc.getPs().println((long)dataTypeId);
    // write dataType
    dc.indent();
    dc.getPs().print("dataType");
    dataType->dump(dc);
    // write DataType_id
    dc.indent();
    dc.getPs().print("DataType_id");
    DataType_id->dump(dc);
    dc.decreaseIndent();
    return retVal;
}


// Getter for dataTypeId
//ui8 CpDataReqMsg::getDataTypeId()
//{
//return this->dataTypeId;
//}

// Setter for dataTypeId
//void CpDataReqMsg::setDataTypeId(ui8 val)
//{
//this->dataTypeId= val;
//}

// Getter for dataType
//org_himalay_msgs_runtime_ByteArray_PTR_TYPE CpDataReqMsg::getDataType()
//{
//return this->dataType;
//}

// Setter for dataType
//void CpDataReqMsg::setDataType(org_himalay_msgs_runtime_ByteArray_PTR_TYPE val)
//{
//this->dataType= val;
//}

// Getter for DataType_id
//org_himalay_msgs_runtime_ByteArray_PTR_TYPE CpDataReqMsg::getDataType_id()
//{
//return this->DataType_id;
//}

// Setter for DataType_id
//void CpDataReqMsg::setDataType_id(org_himalay_msgs_runtime_ByteArray_PTR_TYPE val)
//{
//this->DataType_id= val;
//}

int CpDataReqMsg::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_CpDataReqMsg__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::CpDataReqMsg* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::CpDataReqMsg* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code